### ISSUE TITLE
Performance issue on UHD770

### ISSUE THREAD

**Issue by Silver267** [TavernAI/TavernAI-v1#33]
After extensive tests, I observed that TavernAI is performing badly on both firefox and chromium browsers when using UHD770 as render GPU under 4k (3840x2160) resolution. The consumption is constantly at 100% and the web page becomes laggy too.

**Comment by TavernAI** [comment]
Unfortunately I don't have the opportunity to test it on UHD770. Perhaps someone will be able to find bottlenecks in the code that slow down the work.

**Comment by Bagel7723** [comment]
I see performance issues too, on a Titan Xp I see 15% GPU usage. It seems to only happen while the AI is generating a reply, I think it has something to do with the circular loading animation if I had to guess. I also saw 100% GPU usage a few times, but I have no idea why that was happening. I can only imagine how it might be on UHD graphics if it's this bad on a Titan.
![image](https://user-images.githubusercontent.com/58346379/221417123-64171dcb-82fb-412f-ad0f-77203c02e82f.png)
With the browser tab focused. ^
![image](https://user-images.githubusercontent.com/58346379/221417248-e2cdee37-77b9-4096-ac81-7cf9f0d4e529.png)
With the browser tab unfocused. ^
My browser is Firefox version 102.8.0esr. My resolution is 1920x1080, the browser viewport is 1920×965. I hope this helps.
